### PR TITLE
Added simple one-line (for now) editor example program.

### PR DESCRIPTION
**Pull request by Britefury**

### CODE REVIEW

File: examples/simple_editor.rs [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+extern crate cairo;
+extern crate lspace;
+
+use std::string::String;
+use std::rc::Rc;
+
+use gtk::traits::*;
+use gtk::signal::Inhibit;
+
+// for a list of key names see: http://gtk-rs.org/docs/gdk/enums/key/index.html
```
Comment (ltratt) [inline examples/simple_editor.rs]: Is this comment useful? Surely if someone needs to know this they should look at the `gdk::enums::key` library?

File: examples/simple_editor.rs [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+    }
+}
+
+impl TKeyboardInteractor for LineEditor {
+    fn on_key_event(&self, event: &KeyEvent) {
+        if event.event_type() == KeyEventType::Press {
+            let mut text = self.text_elem.as_text_element().unwrap().get_text().clone();
+            println!("Key val = {}", event.key_val());
+            if event.key_val() == key::BackSpace as u32 {
+                // Backspace
```
Comment (ltratt) [inline examples/simple_editor.rs]: Irrelevant comment given that `key::Backspace` is in the `if` conditional.

File: examples/simple_editor.rs [hunk 3]
```diff
@@ -0,0 +1,110 @@
...
+        Rc::new(LineEditor{text_elem: elem.clone()})
+    }
+}
+
+impl TKeyboardInteractor for LineEditor {
+    fn on_key_event(&self, event: &KeyEvent) {
+        if event.event_type() == KeyEventType::Press {
+            let mut text = self.text_elem.as_text_element().unwrap().get_text().clone();
+            println!("Key val = {}", event.key_val());
+            if event.key_val() == key::BackSpace as u32 {
```
Comment (ltratt) [inline examples/simple_editor.rs]: This feels like it might better be a `match` statement (which will save continually calling `event.key_val()`).

**Comment by Britefury** [comment]
So much for automatic code formatting.

**Comment by ltratt** [comment]
Haha! OK, squash, then I can merge?

**Pull request merged**